### PR TITLE
fix: types static methods and members for Unit class

### PR DESCRIPTION
**Pull request by orelbn**
# Description
- Adds static methods based on documentation: https://mathjs.org/docs/reference/classes/unit.html#unitparsestr--codeunitcode
- Adds static members based on code at `src/type/unit/Unit.js`
- Addresses: https://github.com/josdejong/mathjs/issues/2286

## Test
- Added tests to testTypes.ts

## Additional Notes
I do not see the simplify and format unit in the class documentation. Is it just missing, or am I looking in the wrong place?
- https://mathjs.org/docs/reference/classes/unit.html
- Similarly, the documentation does not reference the following static members.
```
Unit.PREFIXES = PREFIXES
Unit.BASE_DIMENSIONS = BASE_DIMENSIONS
Unit.BASE_UNITS = BASE_UNITS
Unit.UNIT_SYSTEMS = UNIT_SYSTEMS
Unit.UNITS = UNITS
```
- I can take a look at updating the documentation in the future
- Currently, the types I used are pretty generic. I can make them more specific in the future (ex., string[] versus BASE_DIMENSION[])

### CODE REVIEW

**Review by josdejong — COMMENTED** [review]
Thanks a lot Orel, this looks good!

I see that the [public docs of the Unit class](https://mathjs.org/docs/reference/classes/unit.html) that you used are not fully up to date, it would be neat to improve on that. To do that, we need to add some JSDoc annotations to the JavaScript class Unit.js. Is that something you can do too?

**Comment by orelbn** [comment]
> Thanks a lot Orel, this looks good!
> 
> I see that the [public docs of the Unit class](https://mathjs.org/docs/reference/classes/unit.html) that you used are not fully up to date, it would be neat to improve on that. To do that, we need to add some JSDoc annotations to the JavaScript class Unit.js. Is that something you can do too?

Thanks for the feedback; I will make the updates and update the documentation.

File: types/index.d.ts [hunk 1]
```diff
@@ -4014,6 +4015,7 @@ export interface UnitPrefix {
 export interface Unit {
   valueOf(): string
   clone(): Unit
+  _isDerived(): boolean
```
[the remark this answers is not in the document]
Comment (orelbn) [reply types/index.d.ts]: Removed and added `@private`

File: types/index.d.ts [hunk 2]
```diff
@@ -4040,6 +4042,27 @@ export interface Unit {
...
+  PREFIXES: Record<string, UnitPrefix>
+  BASE_DIMENSIONS: string[]
+  BASE_UNITS: Record<string, { dimensions: number[] }>
+  UNIT_SYSTEMS: Record<
+    string,
+    Record<string, { unit: Unit; prefix: UnitPrefix }>
+  >
+  UNITS: Record<string, Unit>
+  parse(str: string): Unit
+  isValuelessUnit(name: string): boolean
```
[the remark this answers is not in the document]
Comment (josdejong) [reply types/index.d.ts:4057]: Thanks!

File: types/index.d.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (josdejong) [reply types/index.d.ts]: Awesome!

**Comment by josdejong** [comment]
Nice, thanks 😎

**Comment by josdejong** [comment]
Fix published now in `v13.0.3`, thanks again!

**Pull request merged**